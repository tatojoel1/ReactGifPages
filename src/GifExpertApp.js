import React, { useState } from "react";
import { Container } from "react-bootstrap";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = ({ defaultCategories = [] }) => {
  //const [categories, setCategories] = useState(["One Punch"]);
  const [categories, setCategories] = useState(defaultCategories);
  return (
    <>
      <Container>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={setCategories} />
        <hr />

        <div>
          {categories.map((category) => (
            <GifGrid key={category} category={category} />
          ))}
        </div>
      </Container>
    </>
  );
};
