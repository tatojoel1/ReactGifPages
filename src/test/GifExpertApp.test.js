import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe("Prueba en el componente GifExpertApp", () => {
  test("El snapshot correspondiente al componente no corresponden", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar una lista de categorias", () => {
    const categories = ["Kimetsu no yaiba", "Dragon ball"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
