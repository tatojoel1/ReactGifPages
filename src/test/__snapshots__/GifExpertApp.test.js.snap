// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el componente GifExpertApp Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <Container
    fluid={false}
  >
    <h2>
      GifExpertApp
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
    <hr />
    <div>
      <GifGrid
        category="Kimetsu no yaiba"
        key="Kimetsu no yaiba"
      />
      <GifGrid
        category="Dragon ball"
        key="Dragon ball"
      />
    </div>
  </Container>
</Fragment>
`;

exports[`Prueba en el componente GifExpertApp El snapshot correspondiente al componente no corresponden 1`] = `
<Fragment>
  <Container
    fluid={false}
  >
    <h2>
      GifExpertApp
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
    <hr />
    <div />
  </Container>
</Fragment>
`;
